### PR TITLE
fix calendar tests, added details test cases, added home page css

### PR DESCRIPTION
**Pull request by jioneyang**
fix calendar tests, added details test cases, added home page css

### CODE REVIEW

File: app/assets/stylesheets/home.scss [hunk 1]
```diff
@@ -1,3 +1,178 @@
...
+    float: right;
+    // left: 0;
+    // position: absolute;
+    // background-color: green;
+}
+
+.half_side .right_side .text {
+    width: 100%;
+    height: 80%;
+    bottom: 0;
```
Comment (chenjoe569) [inline app/assets/stylesheets/home.scss]: CSS code looks good

File: app/controllers/calendars_controller.rb [hunk 2]
```diff
@@ -4,26 +4,30 @@ class CalendarsController < ApplicationController
   # GET /calendars
   # GET /calendars.json
   def index
-    
+          
           if current_user.membership == "Club Member"
               @calendars =Calendar.all.where(:UserId => [current_user.id, nil]).order(:start_time)
+              @admin = false
```
Comment (chenjoe569) [inline app/controllers/calendars_controller.rb]: Refactor this code a little bit. Can put @admin outside of if statement and simply make it true if current_user.membership is admin

File: app/controllers/calendars_controller.rb [hunk 3]
```diff
@@ -72,14 +76,14 @@ def index
...
+    # Use callbacks to share common setup or constraints between actions.
+    def set_calendar
+      @calendar = Calendar.find(params[:id])
+    end
 
-  #   # Never trust parameters from the scary internet, only allow the white list through.
-  #   def calendar_params
-  #     params.require(:calendar).permit(:name, :UserId, :OtherId, :start_time, :end_time, :typeEvent)
-  #   end
+    # Never trust parameters from the scary internet, only allow the white list through.
```
Comment (chenjoe569) [inline app/controllers/calendars_controller.rb]: If you're not using these, delete the comments

File: app/controllers/calendars_controller.rb [hunk 4]
```diff
@@ -62,8 +66,8 @@ def index
   #   end
   # end
 
-  # # DELETE /calendars/1
-  # # DELETE /calendars/1.json
+  # DELETE /calendars/1
```
Comment (chenjoe569) [inline app/controllers/calendars_controller.rb]: if you're not using these, delete the comments or uncomment if it's actually needed

File: app/controllers/calendars_controller.rb [hunk 5]
```diff
@@ -48,8 +52,8 @@ def index
   #   end
   # end
 
-  # # PATCH/PUT /calendars/1
-  # # PATCH/PUT /calendars/1.json
+  # PATCH/PUT /calendars/1
+  # PATCH/PUT /calendars/1.json
   # def update
```
Comment (chenjoe569) [inline app/controllers/calendars_controller.rb]: if you're not using these, delete the comments or uncomment if it's actually needed

File: app/controllers/calendars_controller.rb [hunk 6]
```diff
@@ -32,8 +36,8 @@ def index
   # def edit
   # end
 
-  # # POST /calendars
-  # # POST /calendars.json
+  # POST /calendars
+  # POST /calendars.json
   # def create
```
Comment (chenjoe569) [inline app/controllers/calendars_controller.rb]: if you're not using these, delete the comments or uncomment if it's actually needed

File: app/controllers/calendars_controller.rb [hunk 7]
```diff
@@ -4,26 +4,30 @@ class CalendarsController < ApplicationController
...
-  # def show
-  # end
+  # GET /calendars/1
+  # GET /calendars/1.json
+  def show
+  end
 
-  # # GET /calendars/new
+  # GET /calendars/new
   # def new
```
Comment (chenjoe569) [inline app/controllers/calendars_controller.rb]: if you're not using these, delete the comments or uncomment if it's actually needed

File: app/controllers/calendars_controller.rb [hunk 8]
```diff
@@ -72,14 +76,14 @@ def index
...
+    def set_calendar
+      @calendar = Calendar.find(params[:id])
+    end
 
-  #   # Never trust parameters from the scary internet, only allow the white list through.
-  #   def calendar_params
-  #     params.require(:calendar).permit(:name, :UserId, :OtherId, :start_time, :end_time, :typeEvent)
-  #   end
+    # Never trust parameters from the scary internet, only allow the white list through.
+    # def calendar_params
```
Comment (chenjoe569) [inline app/controllers/calendars_controller.rb]: if you're not using these, delete the comments or uncomment if it's actually needed

File: app/controllers/home_controller.rb [hunk 9]
```diff
@@ -8,16 +8,18 @@ def index
         # redirect_to '/users/sign_in'
         if current_user
             if current_user.membership == "Club Member" or current_user.membership == "Coach"
-                @calendars = Calendar.all.where(:UserId => [current_user.id, nil]).order(:start_time)
+                @calendars = Calendar.all.where(:UserId => [current_user.id, nil]).where("start_time > ?", Time.now.beginning_of_day).order(:start_time)
```
Comment (chenjoe569) [inline app/controllers/home_controller.rb]: Looks good. Tests seem to pass using these

File: app/controllers/home_controller.rb [hunk 10]
```diff
@@ -8,16 +8,18 @@ def index
         # redirect_to '/users/sign_in'
         if current_user
```
Comment (chenjoe569) [inline app/controllers/home_controller.rb]: Don't have to check if current_user. Use before_action authenticate to authenticate user to make sure user is always signed in. Check user_controller for detail

File: app/controllers/user_controller.rb [hunk 11]
```diff
@@ -19,9 +19,9 @@ def booking
   
   def calendar
     if current_user.membership == "Club Member"
-        @calendars = Calendar.all.where(:UserId => [current_user.id, nil]).order(:start_time)
+        @calendars = Calendar.all.where(:UserId => [current_user.id, nil]).where("start_time > ?", Time.now.beginning_of_day).order(:start_time)
```
Comment (chenjoe569) [inline app/controllers/user_controller.rb]: These two lines look exactly the same to me regardless of user membership. Collapse them into one line? Do you even need to check the membership?

File: app/controllers/user_controller.rb [hunk 12]
```diff
@@ -19,9 +19,9 @@ def booking
...
     elsif current_user.membership == "Coach"
-        @calendars = Calendar.all.where(:UserId => [current_user.id, nil]).order(:start_time) #only booked classes currently
+        @calendars = Calendar.all.where(:UserId => [current_user.id, nil]).where("start_time > ?", Time.now.beginning_of_day).order(:start_time) #only booked classes currently
```
Comment (chenjoe569) [inline app/controllers/user_controller.rb]: Look above comment

File: app/views/calendars/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,34 @@
+<p id="notice"><%= notice %></p>
```
Comment (chenjoe569) [inline app/views/calendars/show.html.erb]: Good job. This file looks clean and I get what it's doing pretty easily

File: app/controllers/calendars_controller.rb (hunk 3 above)
Comment (jioneyang) [reply app/controllers/calendars_controller.rb]: Will be using these in future branches

File: app/controllers/calendars_controller.rb (hunk 4 above)
Comment (jioneyang) [reply app/controllers/calendars_controller.rb]: Will be using these in future branches

File: app/controllers/calendars_controller.rb (hunk 5 above)
Comment (jioneyang) [reply app/controllers/calendars_controller.rb]: Will be using these in future branches

File: app/controllers/calendars_controller.rb (hunk 6 above)
Comment (jioneyang) [reply app/controllers/calendars_controller.rb]: Will be using these in future branches

File: app/controllers/home_controller.rb (hunk 10 above)
Comment (jioneyang) [reply app/controllers/home_controller.rb]: I have to check. I've done it without checks and I get a red error page in some cases.

Comment (jioneyang) [reply app/controllers/home_controller.rb]: Because current_user is then nil

Comment (chenjoe569) [reply app/controllers/home_controller.rb]: try before_action :authentica_user! on the top of this file. I think it should work...:P

File: features/payment.feature [hunk 14]
```diff
@@ -17,23 +17,7 @@ And I go to Login page
 
 # Note that I need to specify passwords here because the authentication process won't let me access user password 
 # All scenarios begin assuming no user is logged in yet
-Scenario: Buys 2 group lesson credits
```
Comment (chenjoe569) [inline features/payment.feature]: Why are you deleting these? You can't just delete test you're not passing LOL You should be passing these tests haha.

**Comment by chenjoe569** [comment]
Changes have been made and tests all pass. Most of the code looks refactored. LGTM

**Pull request merged**